src/vm/methodtablebuilder.cpp

```diff
@@ -1489,20 +1489,15 @@ MethodTableBuilder::BuildMethodTableThrowing(
         }
     }
 
-#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
-    if (GetModule()->IsSystem() && !bmtGenerics->HasInstantiation())
+#if defined(FEATURE_HW_INTRINSICS) && (defined(_TARGET_X86_) || defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_))
```

Comment:
Removing `&& !bmtGenerics->HasInstantiation()` may make the methodtablebuilder slower (checking so many candidates). Shall we just add `[Intrinsic]` for methods in the vector classes?

Reply:
I don't think it will, realistically, be that many more candidates. The non-generic candidates are already the vast majority of the checks we do.

CC. @jkotas, however, as he is much more experienced with this and has provided guidance previously.

Reply:
This will not make just the MethodTableBuilder slower. It may also make the JIT slower - all ARM intrinsics will be reported to JIT as intrinsics on x86, and vice versa.

I would mark the platform-neutral types using `[Intrinsic]` until there is an evidence that doing something else is better.

Reply:
The original purpose of this code is to save disk space of mscorlib, which adding `[Intrinsic]` for each HW intrinsic (1300+ now) needs considerable space. But we just have a few of platform-agnostic intrinsic, so using `[Intrinsic]` would be better.

Reply:
> all ARM intrinsics will be reported to JIT as intrinsics on x86, and vice versa.

Ah, right, we need the Target conditional compile as well.

Reply:
> we have the actual types already marked with [Intrinsic] as well, due to the platform ABI handling

Using `[Intrinsic]` on types is different from on methods. We are marking `Vector64/128/256<T>` with `[Intrinsic]` for recognizing these types as `SIMD8/16/32` in JIT, which makes the SIMD type recognizing faster due to avoiding checking so many candidates.

Reply:
Done. Only marking the ones that have been implemented as intrinsics right now.